core/extension/gdextension_manager.cpp

```diff
@@ -210,6 +210,10 @@ void GDExtensionManager::initialize_extensions(GDExtension::InitializationLevel
 	ERR_FAIL_COND(int32_t(p_level) - 1 != level);
 	for (KeyValue<String, Ref<GDExtension>> &E : gdextension_map) {
 		E.value->initialize_library(p_level);
+
+		for (const KeyValue<String, String> &kv : E.value->class_icon_paths) {
+			gdextension_class_icon_paths[kv.key] = kv.value;
+		}
```

Comment:
Is there a better place we can put this? This'll run for each initialization level, which could mean setting the same icon paths over and over.

Or, because icon paths are only need for the editor (I think?) maybe we could do this only on `INITIALIZATION_LEVEL_EDITOR`?